demos/security-trifecta/src/pages/login/LoginCard.tsx

```diff
@@ -6,17 +6,66 @@ import {
   IonCardSubtitle,
   IonCardTitle,
   IonIcon,
+  IonSelect,
+  IonSelectOption,
 } from '@ionic/react';
 import { logInOutline } from 'ionicons/icons';
+import { useEffect, useState } from 'react';
 import { useHistory } from 'react-router-dom';
 import { login } from '../../utils/auth';
+import { UnlockMode } from '../../models/UnlockMode';
+import { getUnlockMode, setUnlockMode, canUseLocking } from '../../utils/session-vault';
+import { Device } from '@ionic-enterprise/identity-vault';
 
 const LoginCard: React.FC = () => {
   const history = useHistory();
+  const [displayLockOptions, setLockOptions] = useState<boolean>(false);
+  const [unlockModes, setUnlockModes] = useState<Array<{ mode: UnlockMode; label: string }>>([]);
```

Comment:
Typically if I define a type manually inline in more than one location, I prefer to hoist it to the top of the file as an interface. Since it's used here and also in `initialize`, if the shape were to change it would need to get updated in multiple places. Certainly a nitpick but I'd add an interface definition at the top of this file and use that:

```
interface UnlockOption {
  mode: UnlockMode;
  label: string;
}

...
const [unlockModes, setUnlockModes] = useState<UnlockOption[]>([]);

...
const initialize = async () => {
  const newUnlockOptions: UnlockOption[] = [
  ...
};
```

Reply:
I agree its a good idea, done.